src/components/data-management/LaunchAnalysisButton.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Button, Tooltip, Popconfirm } from 'antd';
+
+import fileUploadSpecifications from '../../utils/upload/fileUploadSpecifications';
+import UploadStatus from '../../utils/upload/UploadStatus';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import integrationTestConstants from '../../utils/integrationTestConstants';
+import generateGem2sParamsHash from '../../utils/data-management/generateGem2sParamsHash';
+
+import AnalysisModal from './AnalysisModal';
+import { DEFAULT_NA } from '../../redux/reducers/projects/initialState';
+
+const LaunchButtonTemplate = (props) => {
+  // eslint-disable-next-line react/prop-types
+  const { onClick, disabled, text } = props;
+
+  return (
+    <Button
+      data-test-id={integrationTestConstants.ids.LAUNCH_ANALYSIS_BUTTON}
+      type='primary'
+      disabled={disabled}
+      onClick={onClick}
+    >
+      {text}
+    </Button>
+  );
+};
+
+const LaunchAnalysisButton = () => {
+  const experiments = useSelector((state) => state.experiments);
+  const samples = useSelector((state) => state.samples);
+  const backendStatus = useSelector((state) => state.backendStatus);
+
+  const projects = useSelector((state) => state.projects);
+  const { activeProjectUuid } = projects.meta;
+  const activeProject = projects[activeProjectUuid];
+
+  const [gem2sRerunStatus, setGem2sRerunStatus] = useState({ rerun: true, hash: null, reasons: [] });
+  const [analysisModalVisible, setAnalysisModalVisible] = useState(false);
+
+  useEffect(() => {
+    const experimentId = activeProject.experiments[0];
+
+    if (!experiments[experimentId]?.sampleIds.length > 0) return;
+
+    const reasons = [];
+
+    const gem2sStatus = backendStatus[experimentId]?.status.gem2s?.status;
+
+    const gem2sSuccessful = [
+      pipelineStatus.SUCCEEDED, pipelineStatus.RUNNING,
+    ].includes(gem2sStatus);
+
+    const existingParamsHash = backendStatus[experimentId]?.status.gem2s?.paramsHash;
+    const newParamsHash = generateGem2sParamsHash(
+      activeProject,
+      samples,
+      experiments[experimentId],
+    );
+
+    const projectHashEqual = existingParamsHash && existingParamsHash === newParamsHash;
```

Comment:
what is the point of `existingParamsHash && existingParamsHash`?

Reply:
`existingParamsHash` doesn't exist if the project has never been run. So doing `existingParamsHash && ...` will evaluate to false as `existingParamasHash` will be `undefined`. It can be made clearer by setting a default value on line 55. I thnink I'll do that.